src/RamExpression.h

```diff
@@ -187,16 +170,7 @@ class RamUserDefinedOperator : public RamExpression {
         return res;
     }
 
-    void apply(const RamNodeMapper& map) override {
-        for (auto& arg : arguments) {
-            arg = map(std::move(arg));
-        }
-    }
-
 protected:
-    /** Arguments of user defined operator */
-    std::vector<std::unique_ptr<RamExpression>> arguments;
-
```

Comment:
Alter the equal() method calling the super-class and add the check for local state.